src/raglite/_database.py

```diff
@@ -287,6 +288,18 @@ def from_chunks(
         )
 
 
+@lru_cache(maxsize=1)
+def _get_pgvector_version(session: Session) -> str | None:
+    """Get pgvector version.
+
+    Returns
+    -------
+        str | None: Version string if pgvector is installed, None otherwise
+    """
+    result = session.execute(text("SELECT extversion FROM pg_extension WHERE extname = 'vector'"))
+    return result.scalar()
```

Comment:
Suggestion:
- Would return a packagin.version.Version directly, and raise a `RuntimeError` if it's not installed to avoid having a double return type.
- Would drop the docstring as it's a private function and the function name is pretty descriptive.

```suggestion
def _pgvector_version(session: Session) -> Version:
    try:
        result = session.execute(text("SELECT extversion FROM pg_extension WHERE extname = 'vector'"))
        pgvector_version = version.parse(result.scalar())
    except Exception as e:
        error_message = "Unable to parse pgvector version, is pgvector installed?"
        raise ValueError(error_message) from e
    return pgvector_version
```